kalastatic.module

```diff
@@ -73,3 +73,93 @@ function kalastatic_preprocess_html(&$variables) {
     $variables['build_path'] = base_path() . 'kalastatic/';
   }
 }
+
+/**
+* Prepare the variables from a text_with_summary field.
+ */
+function kalastatic_prepare_field__text_with_summary($field, $entity) {
+  $output = [];
+  foreach ($entity->$field->getValue() as $i => $item) {
+    $text = check_markup($item['value'], $item['format']); // TODO: langcode, cache??
+    // TODO: We probably want to have an teaser option that returns the summary.
+    $output[$i] = $text->__toString();
+  }
+
+  // If there's only a single image then don't return a nested array.
+  return count($output) == 1 ? $output[0] : $output;
+}
+
+/**
+ * Prepare the variables from an image field, ready for the image.html.twig
+ */
+function kalastatic_prepare_field__image($field, $entity) {
+  $imgs = [];
+  foreach ($entity->$field->getValue() as $i => $item) {
+    $image_url = $entity->get($field)->entity->uri->value;
+    $imgs[$i] = [
+      'url' => file_create_url($image_url),
+      'title' => $item['title'],
+      'alt' => $item['alt'],
+      'width' => $item['width'],
+      'height' => $item['height'],
+    ];
+  }
+
+  // If there's only a single image then don't return a nested array.
+  return count($imgs) == 1 ? $imgs[0] : $imgs;
+}
+
+/**
+ * Prepare the variables from an image field, ready for the image.html.twig
+ */
+function kalastatic_prepare_field__entity_reference($field, $entity, $formatter) {
+  $output = [];
+  foreach ($entity->$field->referencedEntities() as $i => $item) {
+    if ($formatter) {
+      // Caller wants the referenced entites in a specific format.
+      // TODO: Can we use actual field formatters/view modes somehow?
```

Comment:
I really like the idea of having a 'kalastatic' view mode. That would make it way more obvious in the UI that a particular entity is using KS and not be as mysterious for a non-kalabot.